cluster-api/pkg/controller/machineset/controller.go

```diff
@@ -123,6 +123,16 @@ func (c *MachineSetControllerImpl) createMachine(machineSet *v1alpha1.MachineSet
 		Spec:       machineSet.Spec.Template.Spec,
 	}
 	machine.ObjectMeta.GenerateName = fmt.Sprintf("%s-", machineSet.Name)
+	blockOwnerDeletion := true
+	machine.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+		{
+			Kind: "MachineSet",
+			Name: machineSet.ObjectMeta.GetName(),
+			UID: machineSet.ObjectMeta.GetUID(),
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			BlockOwnerDeletion: &blockOwnerDeletion,
```

Comment:
You should also set the `Controller` field to true. https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L304

Reply:
Added controller bit.